widget/entry.go

```diff
@@ -1523,14 +1520,30 @@ func (e *Entry) setFieldsAndRefresh(f func()) {
 var _ fyne.WidgetRenderer = (*entryRenderer)(nil)
 
 type entryRenderer struct {
-	box, border *canvas.Rectangle
-	scroll      *widget.Scroll
-	icon        *canvas.Image
+	box, border   *canvas.Rectangle
+	scroll        *widget.Scroll
+	icon          *canvas.Image
+	actionWrapper *fyne.Container
 
 	objects []fyne.CanvasObject
 	entry   *Entry
 }
 
+func newEntryRenderer(box, border *canvas.Rectangle, scroll *widget.Scroll, icon *canvas.Image, objects []fyne.CanvasObject, e *Entry) *entryRenderer {
```

Comment:
This looks a bit strange that almost all of the setup code for the entry renderer is outside of this function except for one part of it. I would recommend bringing down the complexity of the function in some other way.

Reply:
huh I'm actually not sure why I did it that way originally, yeah, let me fix that